python/lsst/sims/utils/CodeUtilities.py

```diff
@@ -2,6 +2,38 @@
 import numpy as np
 import numbers
 
+def sims_clean_up():
+    """
+    This method will clean up data caches created by the sims software stack.
+    Any time a cache is added to the sims software stack, it can be added to
+    the list sims_clean_up.targets.  When sims_clean_up() is called, it will
+    loop through the contents of sims_clean_up.targets.  It will call pop()
+    on all of the contents of each sims_clean_up.target, and then reset each
+    sims_clean_up.target to either a blank dict or list (depending on what
+    the target was).
```

Comment:
I am imagining a case in which an actual user (rather than just a unit test) is running our code and finds that it is taking up too much memory.  The user can then call `sims_clean_up()` to delete all of the caches that are currently being used and carry on.  If, later on in the session, memory gets clogged up again, the user will still have the option of calling `sims_clean_up()` to empty the same caches, since `sims_clean_up()` will still be keeping track of all of the caches in play.  Note that objects only get added to `sims_clean_up.targets` when the modules defining them are loaded, so the list should not get gummed up with multiple references to the same object.